src/main/1A5BC.c

```diff
@@ -2574,7 +2610,54 @@ s32 func_800375B4(struct PlayerObj* arg0)
     return 1;
 }
 
-INCLUDE_ASM("asm/us/main/nonmatchings/1A5BC", func_80037708);
+void func_80037708(struct PlayerObj* arg0)
+{
+    s8 temp_s1;
+
+    if (arg0->unk92 == 0) {
+        arg0->unk96 = -1;
+        func_8003795C(arg0);
+        func_80037A24(arg0);
+        temp_s1 = arg0->unk96;
+        if (arg0->unk96 != -1 && arg0->unk98 < D_800F8CCC[arg0->unk96] && (D_800F8CE4[arg0->unk96] == 0 || arg0->unk99 == 0)) {
+            if ((D_800F8C28[arg0->unk96] == 0 || arg0->unk99 == 0 || arg0->unk97 != arg0->unk96)) {
+                if ((temp_s1 != 6 || arg0->unk99 != 0)) {
+                    D_800F8C78[temp_s1](arg0);
+                    arg0->unk97 = temp_s1;
+                    arg0->unk8E = 1;
+                    arg0->unk8F = 1;
+                    arg0->unk90 = 0;
+                    arg0->unk91 = D_800F8C4C[temp_s1 * 2];
```

Comment:
D_800F8C4C and D_800F8C4D are overlapping. This seems to match https://decomp.me/scratch/AY98T

Reply:
This can just be adjusted later, going to merge